src/app/qgisapp.cpp

```diff
@@ -5854,6 +5854,15 @@ void QgisApp::saveMapAsImage()
     QgsMapSettings ms = QgsMapSettings();
     dlg.applyMapSettings( ms );
 
+    //build the expression context
+    QgsExpressionContext expressionContext;
+    expressionContext << QgsExpressionContextUtils::globalScope()
+                      << QgsExpressionContextUtils::projectScope( QgsProject::instance() )
+                      << QgsExpressionContextUtils::mapSettingsScope( ms )
+                      << new QgsExpressionContextScope();
```

Comment:
Oh sorry, that's just me stupidly not cleaning up my copy/pasted code from QgsMapCanvas...

That said, should we attach the expression context in the QgsSaveAsDialog::applyMapSettings( QgsMapSettings & ) function? That'd de-duplicate the code.